src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -81,6 +85,27 @@ public RunWrapper(Run<?,?> build, boolean currentBuild) {
         return r;
     }
 
+    @Whitelisted
+    public List<Cause> getBuildCauses() {
+        if (getRawBuild() != null) {
+            return Collections.unmodifiableList(getRawBuild().getCauses());
+        }
+        else {
+            return new ArrayList<Cause>();
+        }
+    }
+
+    @Whitelisted
+    public List<Cause> getBuildCauses(Class<Cause> causeType) {
+        if (getRawBuild() != null) {
+            return Collections.unmodifiableList(
+                    getRawBuild().getCauses().stream().filter(cause -> causeType.isInstance(cause)).collect(Collectors.toList()));
```

Comment:
probably I'm a bit obsessed, but we could use only once _getRawBuild().getCauses()_

Reply:
agreed, it bothers me too.  Will fix